src/components/Tabs/ScatterPlotTab.tsx

```diff
@@ -86,6 +83,20 @@ export default memo(function ScatterPlotTab(props: ScatterPlotTabProps): ReactEl
   // ^ Memo prevents re-rendering if the props haven't changed.
   const theme = useContext(AppThemeContext);
 
+  const store = useViewerStateStore(
+    useShallow((state) => ({
+      dataset: state.dataset,
+      colorRampRange: state.colorRampRange,
+      colorRamp: state.colorRamp,
+      categoricalPalette: state.categoricalPalette,
+    }))
+  );
+
+  const dataset = useDebounce(store.dataset, 500);
```

Comment:
I know this is not part of the changes, but why did you choose the different values for the debounces?

Reply:
It's a little arbitrary, but `Dataset` has a longer debounce timer so that the scatterplot isn't trying to re-render right as we're loading the new Dataset and the rest of the UI is updating too.